src/ert/storage/local_ensemble.py

```diff
@@ -314,74 +410,164 @@ def _load_single_dataset(
                 f"No dataset '{group}' in storage for realization {realization}"
             ) from e
 
-    def _load_dataset(
+    def _ensure_unified_parameter_dataset_exists(self, group: str):
+        try:
+            self.open_unified_parameter_dataset(group)
+        except FileNotFoundError:
+            self._unify_parameters(group)
+
+    def _ensure_unified_response_dataset_exists(self, group: str):
+        try:
+            self.open_unified_response_dataset(group)
+        except FileNotFoundError:
+            self._unify_responses(group)
+
+    def load_parameters(
         self,
         group: str,
-        realizations: Union[int, npt.NDArray[np.int_], None],
+        realizations: Union[int, Tuple[int], npt.NDArray[np.int_], None] = None,
     ) -> xr.Dataset:
-        if isinstance(realizations, int):
-            return self._load_single_dataset(group, realizations).isel(
-                realizations=0, drop=True
+        self._ensure_unified_parameter_dataset_exists(group)
+
+        try:
+            ds = self.open_unified_parameter_dataset(group)
+
+            if realizations is not None:
+                return ds.sel(
+                    realizations=(
+                        list(realizations)
+                        if type(realizations) is not int
+                        else realizations
+                    )
+                )
+
+            return ds
+        except (ValueError, KeyError, FileNotFoundError) as e:
+            raise KeyError(
+                f"No dataset '{group}' in storage for realization {realizations}"
+            ) from e
+
+    @cached_property
+    def open_gen_data_single_dataset(self) -> xr.Dataset:
```

Comment:
Is the special casing needed?

Reply:
Remnant from doing gen_obs combination, removing it